sdk/src/transaction.rs

```diff
@@ -403,7 +403,9 @@ impl Transaction {
                     .collect();
                 let data = &instruction.data;
                 let e = verify_eth_addresses(data, &instruction_datas);
-                e.map_err(|_| TransactionError::InvalidAccountIndex)?;
```

Comment:
This will need a feature gate to avoid breaking consensus during upgrade

Reply:
Hmm. Is that because the output logs are part of consensus?

Reply:
No. Transaction results are.  The cluster will be vulnerable during upgrade

Reply:
Hmm yes. Probably the team will have to generate the feature to have control over the activation. I'll push a dummy feature for the time being.

Reply:
@t-nelson So, where `verify_precompiles` is used:
in `rpc.rs`:
```
fn verify_transaction(transaction: &Transaction) -> Result<()> {
    ...

    if let Err(e) = transaction.verify_precompiles() {
        return Err(RpcCustomError::TransactionPrecompileVerificationFailure(e).into());
    }

    Ok(())
}
```
Alongside ledger (blockstore processor) and banking stages. The latter already pass `secp256k1_program_enabled` from the bank. So I'll follow the convention and not have the rpc be feature activated, while passing an extra bool to the latter two functions.

I wished that somehow, features instead took the form of global `static`s, so all of this passing around bools could cease. It's an annoyance to write and an annoyance to clean up.

Reply:
> @t-nelson So, where `verify_precompiles` is used:
> in `rpc.rs`:
> 
> ```
> fn verify_transaction(transaction: &Transaction) -> Result<()> {
>     ...
> 
>     if let Err(e) = transaction.verify_precompiles() {
>         return Err(RpcCustomError::TransactionPrecompileVerificationFailure(e).into());
>     }
> 
>     Ok(())
> }
> ```

Shouldn't be too bad.   This function is private and there's a bank available near both call sites

> Alongside ledger (blockstore processor) and banking stages. The latter already pass `secp256k1_program_enabled` from the bank. So I'll follow the convention and not have the rpc be feature activated, while passing an extra bool to the latter two usages.

Sounds like a plan!  If @sakridge still has the private key for the secp256k1 feature, we could even reuse it here.

> I wished that somehow, features instead took the form of global `static`s, so all of this passing around bools could cease. It's an annoyance to write and an annoyance to clean up. Unfortunately, as multiple banks can be simultaneously in existence, this is not a viable solution, to much dismay...

The annoyance is a good source of back pressure against frivolously using features :wink:

Reply:
> If @sakridge still has the private key for the secp256k1 feature, we could even reuse it here.

Don't think we can do this, since that feature is already activated everywhere.

Reply:
It seems a lot for something which isn't critical, is there another appropriate feature this can piggyback on?

Else it will have to be `secp256k1_v2`...

Reply:
Possibly can lump feature with https://github.com/solana-labs/solana/pull/18227 and associated; nonetheless, those are related to syscalls, whereas this PR is for a native program...

Reply:
I guess we can piggyback on this now:
https://github.com/solana-labs/solana/pull/18656

Reply:
I'll wait for the above PR to get merged, then apply the necessary changes here.

Reply:
> I guess we can piggyback on this now:
> #18656

I don't think we'll what to hold up #18656 for this change.  Feel free to use it as a guide.  The interesting bits are the new entry in `sdk/src/feature_set.rs` and helper function in `runtime/src/bank.rs`,  You can then call the bank helper to pass whether the feature has been enabled through to your call sites